File: spec/factories/club_transfer.rb
FactoryBot.define do
  factory :club_transfer do
    player
    old_club factory: %i[club]
    new_club factory: %i[club]
    new_club_name { 'Club Name' }
    start_date { Time.zone.today }
    loan { false }
  end
end

File: app/client/interfaces/ClubTransfer.ts
import { IClub } from "./Club";

export interface IClubTransfer {
  id: number,
  start_date: string,
  old_club: IClub | null,
  old_club_name: string | null,
  new_club: IClub | null,
  new_club_name: string,
  fee: string | null,
  market_value: string | null,
  loan: boolean,
}
